msal/src/test/java/com/microsoft/identity/client/PublicClientApplicationConfigurationTest.java

```diff
@@ -1,19 +1,78 @@
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
 package com.microsoft.identity.client;
 
-import org.junit.Assert;
+import org.junit.Ignore;
 import org.junit.Test;
 
+import static com.microsoft.identity.client.PublicClientApplicationConfiguration.isBrokerRedirectUri;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
 public class PublicClientApplicationConfigurationTest {
+
     @Test
     public void testRedirectUriValidationValid() {
-        Assert.assertTrue(PublicClientApplicationConfiguration.isBrokerRedirectUri("msauth://myPackageName/foo.bar/baz", "myPackageName"));
+        assertTrue(isBrokerRedirectUri("msauth://myPackageName/foo.bar/baz", "myPackageName"));
     }
+
     @Test
     public void testRedirectUriValidationInvalid() {
-        Assert.assertFalse(PublicClientApplicationConfiguration.isBrokerRedirectUri("https://myPackageName/foo.bar/baz", "myPackageName"));
+        assertFalse(isBrokerRedirectUri("https://myPackageName/foo.bar/baz", "myPackageName"));
     }
+
     @Test
     public void testRedirectUriValidationWrongPackage() {
-        Assert.assertFalse(PublicClientApplicationConfiguration.isBrokerRedirectUri("msauth://myPackageName/foo.bar/baz", "notMyPackageName"));
+        assertFalse(isBrokerRedirectUri("msauth://myPackageName/foo.bar/baz", "notMyPackageName"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNullRedirectThrows() {
+        final PublicClientApplicationConfiguration config = new PublicClientApplicationConfiguration();
+        config.setRedirectUri(null);
+        config.validateConfiguration();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEmptyStringRedirectThrows() {
+        final PublicClientApplicationConfiguration config = new PublicClientApplicationConfiguration();
+        config.setRedirectUri("");
+        config.validateConfiguration();
+    }
+
+    @Test
+    @Ignore // Ignore test due to mocking gaps http://g.co/androidstudio/not-mocked
+    public void testValidRedirect() {
+        final PublicClientApplicationConfiguration config = new PublicClientApplicationConfiguration();
+        config.setRedirectUri("msauth://authority");
+        config.validateConfiguration();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    @Ignore // Ignore test due to mocking gaps http://g.co/androidstudio/not-mocked
```

Comment:
This seems... unfortunate.  There should probably be a backlog item for a new approach referenced here.

Reply:
Created:
https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1323372